smart_canvas/core.py

```diff
@@ -1,97 +1,181 @@
 """ core.py """
+from __future__ import annotations
 
 # Default packages
 
 # External packages
 import cv2
-import numpy as np
 from threading import Thread
 import time
+from abc import ABC, abstractmethod
 
 # Internal packages
 from smart_canvas.background import ForegroundMask
 from smart_canvas.gesture_detection import HandDetect
 from smart_canvas.filters.carousel import FilterCarousel
 
-
-class CanvasCore:
+class CanvasCore():
     """
     Class that processes the frame with a dedicated thread.
     """
-
-    def __init__(self, q_consumer):
+    _state = None
+    def __init__(self, q_consumer, screensize: tuple):
```

Comment:
The sanity test is failing here. You should go through all the CanvasCore references and ensure that they all input the screensize argument or remove the argument.